pkg/build/controller/strategy/util.go

```diff
@@ -92,17 +92,17 @@ func mountSecretVolume(pod *kapi.Pod, secretName, mountPath string) {
 
 // setupDockerSecrets mounts Docker Registry secrets into Pod running the build,
 // allowing Docker to authenticate against private registries or Docker Hub.
-func setupDockerSecrets(pod *kapi.Pod, pushSecret string, pullSecret string) {
-	if len(pushSecret) > 0 {
-		mountSecretVolume(pod, pushSecret, DockerPushSecretMountPath)
+func setupDockerSecrets(pod *kapi.Pod, pushSecret, pullSecret *kapi.LocalObjectReference) {
+	if pushSecret != nil {
+		mountSecretVolume(pod, pushSecret.Name, DockerPushSecretMountPath)
 		pod.Spec.Containers[0].Env = append(pod.Spec.Containers[0].Env, []kapi.EnvVar{
 			{Name: "PUSH_DOCKERCFG_PATH", Value: filepath.Join(DockerPushSecretMountPath, "dockercfg")},
```

Comment:
> how you can check that? You mean to inspect the content of the Secret and validate the content to be dockercfg?

Yes.  Retrieve the secret and be sure that it is a dockercfg type: https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/api/types.go#L1882 .  The constant for the key is: https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/api/types.go#L1885